### DISCUSSION TITLE
How do I draw arbitrary things on top of UI?

### DISCUSSION THREAD

**Discussion by nicopap** [bevyengine/bevy#11223] [Q&A] [answered]
This is an often asked question, I've an answer, but it is confined to the depths of the discord server, so I'm reproducing it here for search-engine visibility.

> I want to draw a texture, a 3D mesh, a minimap… anything! on top of the bevy UI. How do I do this?

**Comment by nicopap** [comment, marked as answer]
> As of bevy 0.12

1. Spawn a camera with the [`UiCameraConfig { show_ui: false }`](https://docs.rs/bevy/latest/bevy/prelude/struct.UiCameraConfig.html) component on the same entity.
2. Set the [`Camera::order`](https://docs.rs/bevy/latest/bevy/prelude/struct.Camera.html#structfield.order) field of the camera to something higher than the main one
3. Give the camera a [`RenderLayers`](https://docs.rs/bevy/latest/bevy/render/view/struct.RenderLayers.html) of whatever
4. Give whatever you want to draw on top of UI a `RenderLayers` component intersecting with that of the camera.

**Comment by mirsella** [comment]
Update for bevy 0.14:
- the root ui node should have the [TargetCamera](https://docs.rs/bevy/latest/bevy/prelude/struct.TargetCamera.html) component as `UiCameraConfig` doesn't exist anymore
- on the new camera add `camera2.camera.clear_color = ClearColorConfig::None;` so that it's transparent, and the ui is shown behind. (without it, the screen is gray)

**Reply by extrawurst** [reply]
@mirsella @nicopap would be awesome to have as an example in the bevy repo ❤️

**Reply by mirsella** [reply]
I'll try to do a pr good idea !

**Reply by mirsella** [reply]
https://github.com/bevyengine/bevy/pull/18513
